mevislab.github.io/content/tutorials/dataobjects/contours/contourexample6.md

```diff
@@ -16,112 +16,216 @@ menu:
+{{< highlight filename="CSOLabelRenderer" >}}
+```Python
+labelString = 'Length: ' + str(cso.getLength())
+labelName = 'Distance: '
+labelName += str(cso.getId())
+deviceOffsetX = 0
+deviceOffsetY = 0
+```
+{{</highlight>}}
 
-When connected to a `SoView2DCSOExtensibleEditor`, the `CSOLabelRenderer` module allows access to the currently rendered CSO. Python scripting is used to configure the labels and to define the properties of the label displayed for each CSO in the viewer. *labelString* is set to the unique ID of each CSO, while *labelName* combines the word *name* with the label assigned to the CSO. *deviceOffsetX* and *deviceOffsetY* determine the label's position relative to the CSO, both set to 0 for direct overlay. The labeling displaying the number 1 represent a default label because no specific name is provided for the CSO.
+We are setting the *labelName* to a static text showing the type of the CSO and the unique *ID* of the contour. We also define the *labelString* to the static description of the measurement and the *length* parameter of the CSO.
 
-Replace the default *Label Code* with the following script:
+![labelString and labelName](/images/tutorials/dataobjects/contours/Example6_5.png "labelString and labelName")
 
-{{< highlight filename="" >}}
+You can also round the length by using:
+{{< highlight filename="CSOLabelRenderer" >}}
 ```Python
-if cso.getSubType() == 'distanceLine':
-    labelString = 'Length: ' + str(cso.getLength())
-    labelName = 'Distance '
-else:
-    labelString = ''
-    labelName = ''
-labelName += str(cso.getId())
-labelCaption = labelName + ":  "
-
+labelString = 'Length: ' + str(round(cso.getLength(), 2)) + ' mm'
 ```
 {{</highlight>}}
 
-This script will ensure that the label displays the length of the line for *distanceLine* CSOs and remains empty for other CSO types.
+In order to see all possible parameters of a CSO, add a `CSOInfo` module to your network and connect it to the `CSOManager`. The geometric informations of the selected CSO from `CSOManager` can be seen there.
 
-![Labeled DistanceLine in View2D](/images/tutorials/dataobjects/contours/Ex6_11.png "Labeled DistanceLine in View2D")
+![CSOInfo](/images/tutorials/dataobjects/contours/Ex6_CSOInfo.png "CSOInfo")
 
-In the `CSOLabelRenderer` panel, head to the *Options* tab. Here, you can fine-tune various settings to enhance the appearance of CSO labels. Set the *Label Border Margin X* to *5* to create a margin around selected CSOs. Increase the *Font Size* to *15* for larger and more readable text. Additionally, adjust the *Min Connecting Line Length* to *15* to specify a minimum distance for drawing connecting lines, maintaining a clean and organized layout. 
+For labels shown on greyscale images, it makes sense to add a shadow to your labels. Open the panel of the `SoCSOVisualizationSettings` module and on tab *Misc* check option *Should render shadow*. This increases readability of your labels.
```

Comment:
remove "to your labels"
check option -> check the option
increases readability -> increases the readability